### ISSUE TITLE
[bug] TextInput cursor positioned over hintText

### ISSUE THREAD

**Issue by juliocbcotta** [flutter/flutter#39466]
When using a hintText in an InputDecoration for a TextField, the cursor and the hintText are too close.
<img src="https://user-images.githubusercontent.com/70573/63913981-3ccb0d00-ca08-11e9-8373-6454ad787974.png" width="50%"/>

Search widget code: 
```dart
import 'package:flutter/material.dart';

class SearchWidget extends StatelessWidget {
  final ValueChanged<String> onChanged;

  SearchWidget({this.onChanged});

  @override
  Widget build(BuildContext context) {
    final textStyle = Theme.of(context)
        .primaryTextTheme
        .subhead
        .copyWith(color: Colors.white);

    return Theme(
      data: ThemeData(
        primaryColor: Colors.white,
      ),
      child: Container(
        padding: EdgeInsets.all(8),
        child: TextField(
          minLines: 1,
          autofocus: false,
          onChanged: onChanged,
          cursorColor: Theme.of(context).accentColor,
          style: textStyle,
          decoration: InputDecoration(
              contentPadding: EdgeInsets.all(0),
              hintStyle: textStyle,
              border: OutlineInputBorder(
                borderRadius: BorderRadius.circular(8),
              ),
              prefixIcon: Icon(Icons.search),
              hintText: 'Search',
              alignLabelWithHint: true),
        ),
      ),
    );
  }
}

```

## Proposal

Flutter should move the hintText a few pixels to the right. To match material design guidelines
<img src="https://material.io/components/assets/1Db4T2v9dt2RCOgR4DESgnftJY0pYqXvJ/textfields-outlined-dense.png" width="100%"/>
https://material.io/components/text-fields/#usage

flutter doctor -v
[✓] Flutter (Channel master, v1.9.6-pre.56, on Mac OS X 10.14.6 18G87, locale pt-BR)
    • Flutter version 1.9.6-pre.56 at /Users/juliocotta/flutter/flutter
    • Framework revision c1cb68004a (12 hours ago), 2019-08-28 10:33:43 -0700
    • Engine revision c9d9b5b8f8
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/juliocotta/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/juliocotta/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.3, Build version 10G8
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 38.2.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] IntelliJ IDEA Community Edition (version 2019.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 34.0.4
    • Dart plugin version 191.6183.88

[✓] IntelliJ IDEA Ultimate Edition (version 2019.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 38.2.4
    • Dart plugin version 192.6262.58

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!

**Comment by HansMuller** [comment]
As a workaround, perhaps just use `hintText: ' Search'`

**Comment by BugsBunnyBR** [comment]
Yeah, I did that, but I found no issue to track this problem, so I created this one.

**Comment by jpc-ae** [comment]
This is the same behaviour as on native, except the cursor seems to be drawn underneath the hint text.

I think the proposal should be to make sure the hint text is drawn over the cursor rather than under it.

**Comment by justinmc** [comment]
I'm wondering if this is still how a textfield with a hintText looks nowadays, and if so, what does the Material spec say about this?
